setup.py

```diff
@@ -32,11 +32,18 @@
     ],
     install_requires=[
         'crochet',
-        'service_identity',
         'six',
-        'pyOpenSSL',
         'twisted >= 14.0.0',
         'yelp_bytes',
     ],
+    extras_require={
+        'tls': [
+            # Bug in pip's resolution of extras of extras is broken
+            # so we list twisted[tls] out manually
+            # see https://github.com/pypa/pip/issues/988
+            'pyOpenSSL',
+            'service-identity'
```

Comment:
Sure, looks like we would have been broken before (I just moved the dependencies out of setup_requires into extras_require, didn't actually check against upstream twisted pins).

I've pinned exactly so we are 100% the same as upstream twisted. What's your opinion on the version question?